### PR TITLE
ref(core): Only inject release into entrypoints per default

### PR DESCRIPTION
**Pull request by lforst**
Fixes https://github.com/getsentry/sentry-javascript-bundler-plugins/issues/163

Turns out our approach for injecting the release value sucks in case you want to preserve modules while transpiling because we'll create a throwing `require("\0sentry-release-injector")` statement since `\0sentry-release-injector` is only virtual and you can't "preserve" a virtual file.

This PR is fixing this by slightly changing how the `releaseInjectionTargets` option works. By default we'll inject the release into all entry points now (as opposed to all modules) - without any import magic - just by injecting code. This may have bundle size impact but users can still control how this works by providing a function to `releaseInjectionTargets`.

### CODE REVIEW

File: packages/bundler-plugin-core/src/index.ts [hunk 1]
```diff
@@ -253,15 +219,22 @@ const unplugin = createUnplugin<Options>((options, unpluginMetaContext) => {
...
      * @returns transformed code + source map
      */
-    transform(code, id) {
+    async transform(code, id) {
       logger.debug('Called "transform":', { id });
 
       // The MagicString library allows us to generate sourcemaps for the changes we make to the user code.
       const ms = new MagicString(code);
 
       // Appending instead of prepending has less probability of mucking with user's source maps.
```
Comment (Lms24) [inline packages/bundler-plugin-core/src/index.ts:228]: Is this comment still valid/necessary?

**Review by Lms24 — COMMENTED** [review]
I like this approach. Seems to be a lot simpler than what we had before. I guess this is also kinda "behaviorally" breaking, so let's maybe add a migration note as well? 
lmk when this is ready to ✅

File: packages/bundler-plugin-core/src/index.ts (hunk 1 above)
Comment (Lms24) [reply packages/bundler-plugin-core/src/index.ts:228]: (referring to line 228)

Comment (lforst) [reply packages/bundler-plugin-core/src/index.ts]: Yup thanks! :)

**Comment by lforst** [comment]
@Lms24 should be ready for review!

**Review by Lms24 — APPROVED** [review]
Nice, let's go!

**Comment by pdmshrestha** [comment]
After upgrading from 0.3.0 to 0.4.0, it seems that release is not injected property. 
```
"@sentry/tracing": "^7.38.0",
"@sentry/vite-plugin": "^0.4.0",
"@sentry/vue": "^7.38.0",
```
Didn't have any issue on v0.3.0.

I am using vite to bundle the assets. 
```
import { sentryVitePlugin } from "@sentry/vite-plugin";
```

**Pull request merged**